### ISSUE TITLE
Seeking support

### ISSUE THREAD

**Issue by in0finite** [saul/demofile-net#47]
There should be seeking support in the library.

It seems that demos already have enough data for this to be possible, eg. snapshots on every 60 sec, round-start ticks, etc...

Backward seeking through time should also be possible.

For best usability, I think library should have an API like this:
```cs
DemoParser.SaveToSnapshot(Snapshot s);
DemoParser.RestoreFromSnapshot(Snapshot s);

class Snapshot
{
    List<Entity> entities;
    List<StringTable> stringTables;
    int tick;
    long streamPosition;
    CDemoFileHeader header;

    ...
}
```

**Comment by saul** [comment]
This is definitely do-able, I have some changes on a branch that I need to test that implement seeking to arbitrary ticks.

**Comment by saul** [comment]
WIP PR here: https://github.com/saul/demofile-net/pull/49

**Comment by saul** [comment]
The PR should now be in a state for you to test.

I've tested it locally jumping 77 seconds every 77 seconds, and jumping back 16 seconds every 33 seconds. It seems to read the demo fine.

Can you try it out and let me know how you get on?

**Comment by saul** [comment]
This is now available as v0.12.1
